packages/engine/src/simulation/package/state/packages/behavior_execution/reset_index_col.rs

```diff
@@ -25,7 +19,7 @@ impl IntoArrowChange for ResetIndexCol {
 
         // new_buffer delegates to a method that zeroes the memory so we don't need to initialize
         let mut data = new_buffer::<BehaviorIndexInnerDataType>(num_agents);
-        let mut_data = data.typed_data_mut::<BehaviorIndexInnerDataType>();
+        let _mut_data = data.typed_data_mut::<BehaviorIndexInnerDataType>();
```

Comment:
Pretty sure we can delete this as we don't initialize the memory as per the comment above
```suggestion
```